instrumentation/lettuce/lettuce-4.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/lettuce/v4_0/LettuceSyncClientTest.java

```diff
@@ -0,0 +1,310 @@
+import org.testcontainers.utility.DockerImageName;
+
+@SuppressWarnings("deprecation") // until old http semconv are dropped in 2.0
+class LettuceSyncClientTest {
+
+  @RegisterExtension
+  protected static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  @RegisterExtension static final AutoCleanupExtension cleanup = AutoCleanupExtension.create();
+
+  static final DockerImageName containerImage = DockerImageName.parse("redis:6.2.3-alpine");
+
+  private static final int DB_INDEX = 0;
+
+  // Disable auto reconnect, so we do not get stray traces popping up on server shutdown
+  private static final ClientOptions CLIENT_OPTIONS =
+      new ClientOptions.Builder().autoReconnect(false).build();
+
+  private static final GenericContainer<?> redisServer =
+      new GenericContainer<>(containerImage).withExposedPorts(6379);
+
+  private static String host;
+  private static int port;
+  private static int incorrectPort;
+  private static String dbUriNonExistent;
+  private static String embeddedDbUri;
+
+  private static final ImmutableMap<String, String> testHashMap =
+      ImmutableMap.of(
+          "firstname", "John",
+          "lastname", "Doe",
+          "age", "53");
+
+  static RedisClient redisClient;
+
+  private static StatefulRedisConnection<String, String> connection;
+  static RedisCommands<String, String> syncCommands;
+
+  @BeforeAll
+  public static void setUp() {
+    redisServer.start();
+
+    host = redisServer.getHost();
+    port = redisServer.getMappedPort(6379);
+    embeddedDbUri = "redis://" + host + ":" + port + "/" + DB_INDEX;
+
+    incorrectPort = PortUtils.findOpenPort();
+    dbUriNonExistent = "redis://" + host + ":" + incorrectPort + "/" + DB_INDEX;
+
+    redisClient = RedisClient.create(embeddedDbUri);
+
+    connection = redisClient.connect();
+    syncCommands = connection.sync();
+
+    syncCommands.set("TESTKEY", "TESTVAL");
+    syncCommands.hmset("TESTHM", testHashMap);
+
+    // 2 sets + 1 connect trace
+    testing.clearData();
+  }
+
+  @AfterAll
+  public static void cleanUp() {
+    connection.close();
+    redisServer.stop();
+  }
+
+  @Test
+  void testConnect() {
+    RedisClient testConnectionClient = RedisClient.create(embeddedDbUri);
+    testConnectionClient.setOptions(CLIENT_OPTIONS);
+
+    StatefulRedisConnection<String, String> testConnection = testConnectionClient.connect();
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("CONNECT")
+                        .hasKind(SpanKind.CLIENT)
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(SemanticAttributes.NET_PEER_NAME, host),
+                            equalTo(SemanticAttributes.NET_PEER_PORT, port),
+                            equalTo(SemanticAttributes.DB_SYSTEM, "redis"))));
+    testConnection.close();
```

Comment:
It'd be safer to use the `AutoCleanupExtension`; the connection would get closed even if the assertion fails.